src/onboarding/Onboarding.vue

```diff
@@ -175,9 +108,9 @@ export default {
     this.url.createNewAccount = `${apiUrl}/auth/register?next=%2Fdashboard%2Fsetup_done`;
     this.url.login = `${apiUrl}/dashboard/setup_done`;
 
-    // maybe user redirected from the setup_done page
-    if (this.step === 3) {
-      await this.tryGetUserInfo();
+    if (await this.tryGetUserInfo()) {
+      this.step = 3;
+      return;
```

Comment:
Here, I think we should take into account the case when user's network is slow. In that case, `await this.tryGetUserInfo()` may take up to seconds, which makes the step jumps from 1 to 3.

Also, AFAIK, each time `tryGetUserInfo()` is call, it creates a new API Key. Therefore, I prefer wrapping it inside a condition. What do you think @nguyenkims ?

Reply:
@ngxson the `tryGetUserInfo` shouldn't create a new api key I think, its implementation is on https://github.com/simple-login/app/blob/372466ab067f964ca7bf87d50b1e34b60ef8a58f/app/api/views/user_info.py#L33

Reply:
@nguyenkims Thanks for the confirmation. And for the problem with slow network, I think it's enough to start with a loading state instead of showing step 1 directly.

Reply:
@acasajus @ngxson that's indeed a good point: generally, we should display a loading indicator for any api call. We can further optimize it so it only displays when the call takes more than X seconds. (X to be defined).

Reply:
I think the onboarding page is only shown on install or after the cookie is set. So it shouldn't generate extra tokens either. 
As for the loading, I can leave it in step 1 while the request is being executed and move to 3 afterwards on successful request.